zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbProcessState.java

```diff
@@ -177,9 +183,13 @@ public void deleteProcess(final ProcessRecord processRecord) {
     processColumnFamily.deleteExisting(tenantAwareProcessDefinitionKey);
     processByIdAndVersionColumnFamily.deleteExisting(tenantAwareProcessIdAndVersionKey);
 
-    processesByTenantAndProcessIdAndVersionCache
-        .getOrDefault(processRecord.getTenantId(), new HashMap<>())
-        .remove(processRecord.getBpmnProcessIdBuffer());
+    final var tenantIdAndProcessIdAndVersion =
+        new TenantIdAndProcessIdAndVersion(
+            processRecord.getTenantId(),
+            processRecord.getBpmnProcessIdBuffer(),
+            processRecord.getVersion());
+    processesByTenantAndProcessIdAndVersionCache.invalidate(tenantIdAndProcessIdAndVersion);
```

Comment:
💭 When deleting a process, we now delete the process with the version passed from the cache not all the cached versions.

Reply:
@mustafadagher are you referring to lines 180-182?

```java
processesByTenantAndProcessIdAndVersionCache
    .getOrDefault(processRecord.getTenantId(), new HashMap<>())
    .remove(processRecord.getBpmnProcessIdBuffer());
```

These lines were changed with 618c83fa2b713404ced2b3fedd1e90f4f34e13c1, but kept the same logic (all versions of the process id are removed from the cache). The actual behavior was introduced in 7b56035d61fcf55c28543295ec18186c8e6fb186. As there is no explanation about removing all versions of a process from this cache, I believe it's missed in the review. This could be an easy oversight of calling remove on a Map, although it should've been a remove on a map entry inside the map.

Having said this, we should verify this. The best way to do so that I see is to change this line only (only delete the specific version from the cache) and run all tests.

Reply:
And I think all tests has been green in this PR, so the change is safe.

But also if I update the lines on main to be like follows, it also passes
```java
processesByTenantAndProcessIdAndVersionCache
        .getOrDefault(processRecord.getTenantId(), new HashMap<>())
        .getOrDefault(processRecord.getBpmnProcessIdBuffer(), new Long2ObjectHashMap<>())
        .remove(processRecord.getVersion());
```

Reply:
Thanks for verifying @mustafadagher 🙇